python/shared.py

```diff
@@ -0,0 +1,129 @@
+import json
+from dataclasses import dataclass, field
+from typing import Optional, Iterable, Any
+
+from dataclasses_json import config
+
+TASK_QUEUE = "CustomerLoyaltyTaskQueue"
+CUSTOMER_WORKFLOW_ID_FORMAT = "customer-{}"
+EVENT_HISTORY_THRESHOLD = 10000
+
+# Signal and query names
+SIGNAL_CANCEL_ACCOUNT = "cancelAccount"
+SIGNAL_ADD_POINTS = "addLoyaltyPoints"
+SIGNAL_INVITE_GUEST = "inviteGuest"
+SIGNAL_ENSURE_MINIMUM_STATUS = "ensureMinimumStatus"
+QUERY_GET_STATUS = "getStatus"
+QUERY_GET_GUESTS = "getGuests"
+
+
+@dataclass
+class Customer:
+    customerId: str = field(metadata=config(field_name="customerId"))
+    name: Optional[str] = ""
+    points: Optional[int] = 0
+    guests: Optional[set[str]] = field(default_factory=set)
+    account_active: Optional[bool] = field(default_factory=lambda: True, metadata=config(field_name="accountActive"))
+    status: Optional['Status'] = field(default_factory=lambda: Status(0))
+
+
+@dataclass
+class StatusTier:
+    name: str
+    minimum_points: int
+    guests_allowed: int
+
+
+@dataclass
+class GetStatusResponse:
+    status_level: int
+    tier: StatusTier
+    points: int
+    account_active: bool
+
+
+class Status(Iterable[tuple[Any, Any]]):
+    _level: int
+
+    MEMBER = "Member"
+    BRONZE = "Bronze"
+    SILVER = "Silver"
+    GOLD = "Gold"
+    PLATINUM = "Platinum"
+
+    LEVELS = [
+        StatusTier(name=MEMBER, minimum_points=0, guests_allowed=0),
+        StatusTier(name=BRONZE, minimum_points=500, guests_allowed=1),
+        StatusTier(name=SILVER, minimum_points=1_000, guests_allowed=2),
+        StatusTier(name=GOLD, minimum_points=2_000, guests_allowed=5),
+        StatusTier(name=PLATINUM, minimum_points=5_000, guests_allowed=10),
+    ]
+
+    def __init__(self, level: Optional[int] = 0):
```

Comment:
```suggestion
    def __init__(self, level: int = 0):
```

Why support `None` as a `level`?